include/roaring/roaring.h

```diff
@@ -257,6 +257,33 @@ void roaring_bitmap_andnot_inplace(roaring_bitmap_t *r1,
  */
 void roaring_bitmap_free(const roaring_bitmap_t *r);
 
+/**
+ * A bit of context usable with `roaring_bitmap_add_bulk()`
+ *
+ * Should be initialized with `{0}` (or `memset()` to all zeros).
+ * Callers should treat it as an opaque type
+ */
+typedef struct roaring_bulk_context_s {
+    ROARING_CONTAINER_T *container;
+    int idx;
+    uint16_t key;
+    uint8_t typecode;
+} roaring_bulk_context_t;
+
+/**
+ * Add an item, using context from a previous insert for speed optimization.
+ *
+ * `context` will be used to store information between calls to make bulk loads
+ * faster. `*context` should be zero-initialized before the first call to this
+ * function.
+ *
+ * Modifying the bitmap in any way will invalidate the stored context, calling
+ * this function with a non-zero context after doing any modification invokes
+ * undefined behavior.
+ */
```

Comment:
The user should be informed about how to exploit the optimization IMO. That means stating it comes into play when several values sharing the same key are inserted successively.
If purely random and sparse keys are being obtained from a stream, for example, explicitly calling the function is unlikely to help in any meaningful way, while involving some extra boilerplate.